### PR TITLE
Rule for equals on array

### PR DESCRIPTION
**Pull request by johanandren**
Ported from scapegoat, checks for equals comparison between two arrays, which in reality is an object identity comparison and probably not what you intend if you use ==

### CODE REVIEW

**Comment by dragos** [comment]
Sorry for the long delay in getting back to the PR queue... it seems that there are merge conflicts (probably due to other PRs being merged...) could you please rebase?

**Comment by johanandren** [comment]
Sorry right back at you, busy times :) Now it is rebased with the latest master.

File: rules/core/src/main/scala/com/typesafe/abide/core/ArrayEquals.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class ArrayEquals(val context: Context) extends WarningRule {
+
+  import context.universe._
+
+  val name = "array-equals"
+
+  case class Warning(appl: Tree) extends RuleWarning {
+    val pos = appl.pos
+    val message = "equals for Arrays is not a deep equals and will always perform instance equality"
```
Comment (wookietreiber) [inline rules/core/src/main/scala/com/typesafe/abide/core/ArrayEquals.scala]: Would be nice to suggests `sameElements`, e.g.:

```scala
val message = "equals for Arrays is not a deep equals and will always perform instance equality (see the implicitly available ArrayOps.sameElements for a deep equals)"
```

**Comment by SethTisue** [comment]
I'm tentatively closing old abandoned-seeming PRs. We can reopen if someone wants to move it forward again (after a rebase).

**Pull request closed without merging**